services/src/main/java/org/fao/geonet/api/records/MetadataValidateApi.java

```diff
@@ -136,6 +153,26 @@ Reports validateRecord(
 
         Locale locale = languageUtils.parseAcceptLanguage(request.getLocales());
 
+        boolean isSubtemplate = metadata.getDataInfo().getType() == MetadataType.SUB_TEMPLATE;
+        boolean validSet = (isvalid != null);
+        if (!isSubtemplate && validSet) {
+            throw new BadParameterEx("isvalid", "<not Null for not subTemplate>");
```

Comment:
In the API, we tried so far (I think) to have more meaningful error message for end users eg. https://github.com/geonetwork/core-geonetwork/blob/develop/services/src/main/java/org/fao/geonet/api/records/MetadataSocialApi.java#L132

Same below l162